Index: default.py
===================================================================
--- default.py	(révision 235)
+++ default.py	(copie de travail)
@@ -27,6 +27,18 @@
 if not xbmcvfs.exists(DATA_PATH):
     xbmcvfs.mkdir(DATA_PATH)
 
+def _unicode( text, encoding='utf-8' ):
+    try: text = unicode( text, encoding )
+    except: pass
+    return text
+
+
+import unicodedata
+def normalize_string( text ):
+    try: text = unicodedata.normalize( 'NFKD', _unicode( text ) ).encode( 'ascii', 'ignore' )
+    except: pass
+    return text
+
 def log(msg):
     xbmc.log( str( msg ),level=xbmc.LOGDEBUG )
 
@@ -149,6 +161,9 @@
                     DIALOG_PROGRESS.close()
                     xbmcgui.Dialog().ok(__language__(32103),__language__(32104))
                     break
+            #change tuple format to list. because 'tuple' object does not support item assignment
+            show = list(show)
+            show[ 0 ] = normalize_string( show[ 0 ] )
             log( "### %s" % show[0] )
             current_show["localname"] = show[0]
             current_show["path"] = show[1]
