diff --git a/xbmc/interfaces/python/xbmcmodule/xbmcguimodule.cpp b/xbmc/interfaces/python/xbmcmodule/xbmcguimodule.cpp
index 1692a5d..40e3f74 100644
--- a/xbmc/interfaces/python/xbmcmodule/xbmcguimodule.cpp
+++ b/xbmc/interfaces/python/xbmcmodule/xbmcguimodule.cpp
@@ -29,6 +29,7 @@
 #include "pyutil.h"
 #include "action.h"
 #include "utils/log.h"
+#include "input/MouseStat.h"
 #include "guilib/GUIWindowManager.h"
 #include "guilib/GUIListItem.h"
 
@@ -99,12 +100,41 @@ namespace PYXBMC
     return Py_BuildValue((char*)"i", id);
   }
 
+  // getMousePosition() method
+  PyDoc_STRVAR(getMousePosition__doc__,
+    "getMousePosition() -- Returns the current mouse position as a x,y integer tuple.\n"
+    "\n"
+    "example:\n"
+    "  - pos = xbmcgui.getMousePosition()\n");
+
+  PyObject* XBMCGUI_GetMousePosition(PyObject *self, PyObject *args)
+  {
+    PyXBMCGUILock();
+    // This code comes from windows/GUIWindowDebugInfo.cpp
+    CGUIWindow *window = g_windowManager.GetWindow(g_windowManager.GetFocusedWindow());
+    CGUIWindow *pointer = g_windowManager.GetWindow(WINDOW_DIALOG_POINTER);
+    CPoint point;
+    if (pointer)
+      point = CPoint(pointer->GetXPosition(), pointer->GetYPosition());
+    if (window)
+    {
+      // transform the mouse coordinates to this window's coordinates
+      g_graphicsContext.SetScalingResolution(window->GetCoordsRes(), true);
+      point.x *= g_graphicsContext.GetGUIScaleX();
+      point.y *= g_graphicsContext.GetGUIScaleY();
+      g_graphicsContext.SetRenderingResolution(g_graphicsContext.GetResInfo(), false);
+    }
+    PyXBMCGUIUnlock();
+    return Py_BuildValue((char*)"(ll)", (int)point.x, (int)point.y);
+  }
+
   // define c functions to be used in python here
   PyMethodDef xbmcGuiMethods[] = {
     {(char*)"lock", (PyCFunction)XBMCGUI_Lock, METH_VARARGS, lock__doc__},
     {(char*)"unlock", (PyCFunction)XBMCGUI_Unlock, METH_VARARGS, unlock__doc__},
     {(char*)"getCurrentWindowId", (PyCFunction)XBMCGUI_GetCurrentWindowId, METH_VARARGS, getCurrentWindowId__doc__},
     {(char*)"getCurrentWindowDialogId", (PyCFunction)XBMCGUI_GetCurrentWindowDialogId, METH_VARARGS, getCurrentWindowDialogId__doc__},
+    {(char*)"getMousePosition", (PyCFunction)XBMCGUI_GetMousePosition, METH_VARARGS, getMousePosition__doc__},
     {NULL, NULL, 0, NULL}
   };
 
